src/blas/backends/rocblas/rocblas_helper.hpp

```diff
@@ -205,6 +205,66 @@ inline rocblas_side get_rocblas_side_mode(oneapi::mkl::side lr) {
     }
 }
 
+template <typename T>
+inline rocblas_datatype get_rocblas_datatype() {
+    static_assert(false);
+}
+
+template <>
+inline rocblas_datatype get_rocblas_datatype<rocblas_half>() {
+    return rocblas_datatype_f16_r;
```

Comment:
Was there a reason for these not to be `constexpr`? I don't think the code needs to be changed since these are being used in a very limited scope, but I'm curious.

Reply:
Short answer: just the way the code was written for current requirements.